core/src/main/scala/cats/syntax/option.scala

```diff
@@ -394,4 +396,9 @@ object OptionOps {
     def apply[E](ifEmpty: => E)(implicit F: ApplicativeError[F, _ >: E]): F[A] =
       ApplicativeError.liftFromOption(oa, ifEmpty)
   }
+
+  final private[syntax] class ToInvalidAPartiallyApplied[F[_], A](oa: Option[A]) {
```

Comment:
Would you try to benefit from `AnyVal` here please?
```suggestion
  final private[syntax] class ToInvalidAPartiallyApplied[F[_], A](private val oa: Option[A]) extends AnyVal {
```
I'm not 100% there's no another hurdle, but I think if it worked then it would be nice to have.